src/Domain/LeanCode.DomainModels/Ids/ITypedId.cs

```diff
@@ -0,0 +1,36 @@
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+
+namespace LeanCode.DomainModels.Ids;
+
+[SuppressMessage("?", "CA1000", Justification = "Roslyn bug.")]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public interface IPrefixedTypedId<TSelf> : IEquatable<TSelf>, IComparable<TSelf>
```

Comment:
Can we also add and implement `ISpanFormattable` and `ISpanParsable`? They are very useful when performing string interpolation/formatting (and they would be even more useful if we used any stringlike type (Ulid? :stuck_out_tongue:) that isn't directly supported by STJ).

Reply:
Next version, this one is already big.